libsplinter/src/circuit/template/mod.rs

```diff
@@ -540,4 +648,29 @@ rules:
         file.write_all(data)
             .expect("Error writing example template yaml.");
     }
+
+    fn verify_example_yaml_string(yaml: String) {
+        // Validate the main headers in the YAML string
+        assert!(yaml.contains("version: v1"));
```

Comment:
I would suggest also adding:

```
let _: serde_yaml::Value = serde_yaml::from_str(&yaml).expect("Invalid yaml was returned");
```
That way, you validate that it is actually parseable 

(likewise, you could then inspect the contents of that `Value`, but that seems like a lot of effort for something similar to your `contains` calls).

Reply:
I didn't initially include this because the `load_raw_yaml` is just returning the `serde_yaml::to_string` method so that seemed redundant, but now I see this being useful in the future if that method is ever changed so I included the line you suggested. I left the string inspection with the `contains` calls as-is.